core/src/nucleus/ribosome/api/mod.rs

```diff
@@ -237,18 +257,30 @@ pub fn call(
             field_name: &str,
             _signature: &Signature,
         ) -> Result<FuncRef, InterpreterError> {
-            // Take the canonical name and find the corresponding ZomeApiFunction index
-            let index = ZomeApiFunction::str_to_index(&field_name);
-            match index {
-                index if index == ZomeApiFunction::MissingNo as usize => {
-                    return Err(InterpreterError::Function(format!(
-                        "host module doesn't export function with name {}",
-                        field_name
-                    )));
-                }
+            let api_fn = ZomeApiFunction::from_str(&field_name)?;
```

Comment:
This is the line causing a compile error

<img width="913" alt="screen shot 2018-09-14 at 12 45 42 pm" src="https://user-images.githubusercontent.com/1409121/45564015-158e2680-b81d-11e8-8117-3f67a8abdcd9.png">

Has to do with this, but I'm not sure how to fix:
https://paritytech.github.io/wasmi/wasmi/trait.HostError.html

Reply:
Don't know much about the codebase yet but it seems like there is specific type of trait that string should implement for the str type. I would 1) check if it something is derivable 2) write an implementation definition of the trait for str 3)check if an implementation already exists and bring it into scope.

Reply:
Thanks Ashanti
As it turns out, I was misled by the error... the problem really had to do with the fact that `from_str` was returning a "Result" type value, with an Ok or an Err, whereas before it had been something different. That was messing up the `match` statement underneath